data-import.qmd

```diff
@@ -64,165 +48,165 @@ read_csv("data/students.csv") |>
+Usually `read_csv()` uses the first line of the data for the column names, which is a very common convention.
+But sometime there are a few lines of metadata at the top of the file.
+You can use `skip = n` to skip the first `n` lines or use `comment = "#"` to drop all lines that start with (e.g.) `#`:
 
 ```{r}
 #| message: false
 
-students <- read_csv("data/students.csv", na = c("N/A", ""))
+read_csv(
+  "The first line of metadata
+  The second line of metadata
+  x,y,z
+  1,2,3",
+  skip = 2
+)
 
-students
+read_csv(
+  "# A comment I want to skip
+  x,y,z
+  1,2,3",
+  comment = "#"
+)
 ```
 
-Once you read data in, the first step usually involves transforming it in some way to make it easier to work with in the rest of your analysis.
-For example, the column names in the `students` file we read in are formatted in non-standard ways.
-You might consider renaming them one by one with `dplyr::rename()` or you might use the `janitor::clean_names()` function turn them all into snake case at once.[^data-import-1]
-This function takes in a data frame and returns a data frame with variable names converted to snake case.
-
-[^data-import-1]: The [janitor](http://sfirke.github.io/janitor/) package is not part of the tidyverse, but it offers handy functions for data cleaning and works well within data pipelines that uses `|>`.
+In other cases, the data might not have column names.
+You can use `col_names = FALSE` to tell `read_csv()` not to treat the first row as headings, and instead label them sequentially from `X1` to `Xn`:
 
 ```{r}
 #| message: false
 
-library(janitor)
-students |>
-  clean_names()
+read_csv(
+  "1,2,3
+  4,5,6",
+  col_names = FALSE
+)
 ```
 
-Another common task after reading in data is to consider variable types.
-For example, `meal_type` is a categorical variable with a known set of possible values.
-In R, factors can be used to work with categorical variables.
-We can convert this variable to a factor using the `factor()` function.
-You'll learn more about factors in @sec-factors.
+Alternatively you can pass `col_names` a character vector which will be used as the column names:
 
 ```{r}
-students <- students |>
-  clean_names() |>
-  mutate(meal_plan = factor(meal_plan))
+#| message: false
 
-students
+read_csv(
+  "1,2,3
+  4,5,6",
+  col_names = c("x", "y", "z")
+)
 ```
 
-Note that the values in the `meal_type` variable has stayed exactly the same, but the type of variable denoted underneath the variable name has changed from character (`<chr>`) to factor (`<fct>`).
-
-Before you move on to analyzing these data, you'll probably want to fix the `age` column as well: currently it's a character variable because of the one observation that is typed out as `five` instead of a numeric `5`.
-We discuss the details of fixing this issue in @sec-import-spreadsheets in further detail.
-
-### Compared to base R
+These arguments are all you need to know to read \~80% of CSV files that you'll encounter in practice.
```

Comment:
Since 80% is likely a made-up number, how about "a large majority of CSV files" instead?